File: src/main/java/me/soo/helloworld/controller/UserController.java
package me.soo.helloworld.controller;

import lombok.RequiredArgsConstructor;
import me.soo.helloworld.annotation.CurrentUser;
import me.soo.helloworld.annotation.LoginRequired;
import me.soo.helloworld.model.user.*;
import me.soo.helloworld.service.LoginService;
import me.soo.helloworld.service.PushNotificationService;
import me.soo.helloworld.service.UserService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;

import static me.soo.helloworld.util.http.HttpResponses.HTTP_RESPONSE_CONFLICT;
import static me.soo.helloworld.util.http.HttpResponses.HTTP_RESPONSE_OK;

@RestController
@RequiredArgsConstructor
@RequestMapping("/users")
public class UserController {

    private final UserService userService;

    private final LoginService loginService;

    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping("/signup")
    public void userSignUp(@Valid @RequestBody User user) {
        userService.userSignUp(user);
    }

    @GetMapping("/id-check")
    public ResponseEntity<Void> isUserIdDuplicate(@RequestParam String userId) {
        if (userService.isUserIdExist(userId)) {
            return HTTP_RESPONSE_CONFLICT;
        }

        return HTTP_RESPONSE_OK;
    }

    @PostMapping("/login")
    public void userLogin(@Valid @RequestBody UserLoginRequest loginRequest) {
        loginService.login(loginRequest);
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @GetMapping("/logout")
    public void userLogout() {
        loginService.logout();
    }

    @PostMapping("/password-finder")
    public void userFindPassword(@Valid @RequestBody UserFindPasswordRequest findPasswordRequest) {
        userService.findUserPassword(findPasswordRequest);
    }
}

File: src/main/java/me/soo/helloworld/controller/UserController.kt
package me.soo.helloworld.controller

import me.soo.helloworld.model.user.FindPasswordRequest
import me.soo.helloworld.model.user.LoginRequest
import me.soo.helloworld.model.user.User
import me.soo.helloworld.service.LoginService
import me.soo.helloworld.service.UserService
import me.soo.helloworld.util.http.HttpResponse
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
@RequestMapping("/users")
class UserController(

    private val userService: UserService,

    private val loginService: LoginService
) {

    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping("/signup")
    fun signUp(@Valid @RequestBody user: User) = userService.signUp(user)

    @GetMapping("/id-check")
    fun checkDuplicateId(@RequestParam userId: String) =
        if (!userService.isUserIdExist(userId)) HttpResponse.OK else HttpResponse.CONFLICT

    @PostMapping("/login")
    fun login(@Valid @RequestBody loginRequest: LoginRequest) = loginService.login(loginRequest)

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @GetMapping("/logout")
    fun logout() = loginService.logout()

    @PostMapping("/pw-finder")
    fun findPassword(@Valid @RequestBody findPasswordRequest: FindPasswordRequest) =
        userService.findPassword(findPasswordRequest)
}
